### PR TITLE
[CATEGORY] Endpoint search  UserById

### PR DESCRIPTION
**Pull request by Jair-Academlo**
## Description
AS: users 
I WANT TO get an user by id
SO I can see information the user by id

Acceptance Criteria:

A GET request must be made to the path /users/:id

The endpoint must return the user data. If it doesn't exist, it should return an error.

## Evidence:

search success!

userById

in case error

error_userByID

### CODE REVIEW

File: controllers/users/userById.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const { endpointResponse } = require("../../helpers/success");
+const { ErrorObject } = require("../../helpers/error");
+const { User } = require("../../database/models");
+
+module.exports = {
+  getById: catchAsync(async (req, res, next) => {
+    try {
+      const { id } = req.params;
+
+      const userId = await User.findOne({ where: { id } });
```
Comment (miguelbel00) [inline controllers/users/userById.js:12]: userId can be replace for userFound

File: controllers/users/userById.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      if (!userId) {
+        return next(new ErrorObject(" Id not found ", 400));
```
Comment (miguelbel00) [inline controllers/users/userById.js:15]: replace only for throw new Error.....

File: package.json [hunk 3]
```diff
@@ -26,14 +26,16 @@
...
-    "nodemon": "^2.0.15",
     "sequelize": "^6.19.0",
     "sequelize-cli": "6.2.0"
   },
   "devDependencies": {
     "chai": "^4.3.4",
     "jest": "27.0.6",
     "mocha": "^9.1.3",
+    "nodemon": "^2.0.20",
+    "pg": "^8.8.0",
```
Comment (miguelbel00) [inline package.json:37]: this package it's not necessary, npm remove pg

File: package.json [hunk 4]
```diff
@@ -26,14 +26,16 @@
...
+    "pg-hstore": "^2.3.4",
```
Comment (miguelbel00) [inline package.json:38]: this package it's not necessary, npm remove pg-hstore

File: routes/index.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
 const express = require('express')
 const categoryRoutes = require('./categoryRoutes')
+const users = require("./user.routes")
 const router = express.Router()
 
-router.use('/', categoryRoutes)
+router.use('/', categoryRoutes, users)
```
Comment (miguelbel00) [inline routes/index.js:6]: remove categorysRoutes, only put users like controller

**Review by miguelbel00 — CHANGES_REQUESTED** [review]
readme file have comments unnecessary,
one feature created per PR

**Comment by miguelbel00** [comment]
Resuelve las inconsistencias que te marque anteriormente

File: controllers/users/userById.js (hunk 1 above)
Comment (Jair-Academlo) [reply controllers/users/userById.js]: The changes are already resolved.

File: package.json (hunk 3 above)
Comment (Jair-Academlo) [reply package.json:37]: I have problems using mysql at the moment I am using postgres, the packages can be removed at the end when the project is going to be released to production.

File: package.json (hunk 4 above)
Comment (Jair-Academlo) [reply package.json:38]: I have problems using mysql at the moment I am using postgres, the packages can be removed at the end when the project is going to be released to production.

File: routes/index.js (hunk 5 above)
Comment (Jair-Academlo) [reply routes/index.js]: the changes are already resolved

File: controllers/users/userById.js (hunk 2 above)
Comment (Jair-Academlo) [reply controllers/users/userById.js]: the changes are already resolved

**Pull request closed without merging**